src/binding/c/rma_api.txt

```diff
@@ -0,0 +1,466 @@
+    .seealso: MPI_Win_flush_all MPI_Win_flush_local MPI_Win_flush_local_all MPI_Win_lock MPI_Win_lock_all
+    .impl: mpid
+    .earlyreturn: pt2pt_proc_null
+
+MPI_Win_flush_all:
+    .desc: Complete all outstanding RMA operations at all targets
+    .seealso: MPI_Win_flush MPI_Win_flush_local MPI_Win_flush_local_all MPI_Win_lock MPI_Win_lock_all
+    .impl: mpid
+
+MPI_Win_flush_local:
+    .desc: Complete locally all outstanding RMA operations at the given target
+    .seealso: MPI_Win_flush MPI_Win_flush_all MPI_Win_flush_local_all MPI_Win_lock MPI_Win_lock_all
+    .impl: mpid
+    .earlyreturn: pt2pt_proc_null
+
+MPI_Win_flush_local_all:
+    .desc: Complete locally all outstanding RMA operations at all targets
+    .seealso: MPI_Win_flush MPI_Win_flush_all MPI_Win_flush_local MPI_Win_lock MPI_Win_lock_all
+    .impl: mpid
+
+MPI_Win_free:
+    .desc: Free an MPI RMA window
+{
+    if (MPIR_Process.attr_free && win_ptr->attributes) {
+        mpi_errno = MPIR_Process.attr_free(win_ptr->handle, &win_ptr->attributes);
+    }
+    /*
+     * If the user attribute free function returns an error,
+     * then do not free the window
+     */
+    if (mpi_errno != MPI_SUCCESS)
+        goto fn_fail;
+
+    /* We need to release the error handler */
+    if (win_ptr->errhandler && !(HANDLE_IS_BUILTIN(win_ptr->errhandler->handle))) {
+        int in_use;
+        MPIR_Errhandler_release_ref(win_ptr->errhandler, &in_use);
+        if (!in_use) {
+            MPIR_Handle_obj_free(&MPIR_Errhandler_mem, win_ptr->errhandler);
+        }
+    }
+
+    mpi_errno = MPID_Win_free(&win_ptr);
+    if (mpi_errno)
+        goto fn_fail;
+    *win = MPI_WIN_NULL;
+}
+
+MPI_Win_get_group:
+    .desc: Get the MPI Group of the window object
+{
+    MPIR_Group *group_ptr = NULL;
+    MPIR_Comm *win_comm_ptr = win_ptr->comm_ptr;
+    mpi_errno = MPIR_Comm_group_impl(win_comm_ptr, &group_ptr);
+    if (mpi_errno != MPI_SUCCESS)
+        goto fn_fail;
+    *group = group_ptr->handle;
+}
+
+MPI_Win_get_info:
+    .desc: Returns a new info object containing the hints of the window
+    .seealso: MPI_Win_set_info
+    .impl: mpid
+/*
+    Notes:
+    
+    The info object returned in info_used will contain all hints currently active
+    for this window. This set of hints may be greater or smaller than the set of
+    hints specified when the window was created, as the system may not recognize
+    some hints set by the user, and may recognize other hints that the user has not
+    set.
+*/
+
+MPI_Win_get_name:
+    .desc: Get the print name associated with the MPI RMA window
+{
+    MPL_strncpy(win_name, win_ptr->name, MPI_MAX_OBJECT_NAME);
+    *resultlen = (int) strlen(win_name);
+}
+
+MPI_Win_lock:
+    .desc: Begin an RMA access epoch at the target process
+    .skip: validate-ASSERT, validate-LOCK_TYPE
+    .impl: mpid
+    .earlyreturn: pt2pt_proc_null
+{ -- error_check --
+    CHECKMASK: assert, assert, MPI_MODE_NOCHECK
```

Comment:
Note is lost in `win_lock.c`, `win_post.c`, `win_start.c`.

Reply:
Added.